src/config.rs

```diff
@@ -558,26 +554,27 @@ impl Config {
       }
     };
 
-    let shell_args = if matches.is_present(arg::CLEAR_SHELL_ARGS) {
-      Vec::new()
-    } else {
+    let shell_args = if matches.is_present(arg::SHELL_ARG) {
       matches
         .values_of(arg::SHELL_ARG)
         .unwrap()
         .map(str::to_owned)
         .collect()
+    } else {
+      Vec::new()
     };
 
-    let shell_present = matches.occurrences_of(arg::CLEAR_SHELL_ARGS) > 0
-      || matches.occurrences_of(arg::SHELL) > 0
+    let shell_present = matches.occurrences_of(arg::SHELL) > 0;
+
+    let shell_args_present = matches.occurrences_of(arg::CLEAR_SHELL_ARGS) > 0
       || matches.occurrences_of(arg::SHELL_ARG) > 0;
 
     Ok(Self {
       check: matches.is_present(arg::CHECK),
       dry_run: matches.is_present(arg::DRY_RUN),
       dump_format: Self::dump_format_from_matches(matches)?,
       highlight: !matches.is_present(arg::NO_HIGHLIGHT),
-      shell: matches.value_of(arg::SHELL).unwrap().to_owned(),
+      shell: matches.value_of(arg::SHELL).unwrap_or("").to_owned(),
```

Comment:
`""` shouldn't be used to indicated that the argument isn't present, since the user could actually pass the empty string as the argument. We should use an `Option` for the `shell` field.

Reply:
This might also let us get rid of `shell_present`, which would be nice.